bin/server.rs

```diff
@@ -111,6 +129,55 @@ async fn main() -> anyhow::Result<()> {
                 .join("bindle")
         });
 
+    // find bindle directory
+    //   1. cli options if set
+    //   2. config file if set
+    //   3. default
+    //   4. hardcoded `./.bindle/keyring.toml`
+    // TODO: Should we ensure a keyring?
+    let keyring_file: PathBuf = opts
+        .keyring_file
+        .or_else(|| {
+            config
+                .get("keyring")
+                .map(|v| v.as_str().unwrap().parse().unwrap())
+        })
+        .unwrap_or(PathBuf::from(default_config_dir().join("keyring.toml")));
+
+    // We might want to do something different in the future. But what we do here is
+    // load the file if we can find it. If the file just doesn't exist, we print a
+    // warning and load a placeholder. This prevents the program from failing when
+    // a keyring does not exist.
+    //
+    // All other cases are considered errors worthy of failing.
+    let keyring: KeyRing = match std::fs::metadata(&keyring_file) {
+        Ok(md) if md.is_file() => load_toml(keyring_file).await?,
+        Ok(_) => Err(anyhow::anyhow!(
+            "Expected {} to be a regular file",
+            keyring_file.display()
+        ))?,
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
+            println!("No keyring.toml found. Using default keyring.");
```

Comment:
This should probably be a `warn!`

Reply:
I changed all of the `println!`s in server.rs to `warn!`s